waku/node/waku_node.nim

```diff
@@ -344,6 +344,17 @@ proc publish*(
   
   return ok()
 
+proc messageSizeValidator(topic: string, message: WakuMessage): Future[pubsub.ValidationResult] {.async.} =
+  trace "message size validator is called"
+
+  let messageSizeBytes = uint64(message.encode().buffer.len)
+
+  if(messageSizeBytes > MaxWakuMessageSize):
+      trace fmt"Message size exceeded maximum of {DefaultMaxWakuMessageSizeStr}"
```

Comment:
In the future we might want to use metrics instead so that we can get some stats on why messages are being rejected

Reply:
Makes sense! Not sure I understand instead of what though. Wouldn't it just be adding a line of code updating the metrics?

Reply:
> Makes sense! Not sure I understand instead of what though. Wouldn't it just be adding a line of code updating the metrics?

Yes exactly :) I was thinking of having something similar to the following, where we can count the number of errors by its type:

https://github.com/waku-org/nwaku/blob/c1121dd1db27344ad38d6ba85c072f0e75eb4e8e/waku/node/peer_manager/peer_manager.nim#L109

To be considered as a future enhancement :)